### PR TITLE
Docs 2

### PR DESCRIPTION
**Pull request by TheMcMurder**
Next steps [outlined in previous MR](https://github.com/single-spa/single-spa-layout/pull/18#discussion_r425395632):
- ~Document the options that each object can have during constructRoute phase~
- Document the HTML syntax option
- Document constructApplications

This MR completes documenting the options that each object can have during `constructRoute` phase.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,16 +37,56 @@ const layoutEngine = constructLayoutEngine({routes, applications})
...
+3. [Route](#Route)
+
+##### Router
+
+The root object/node. Router can have the following options:
+
+- mode: 'history' | 'hash'
+- base: base url where the router is active
+- containerEl: which cssSelector is where the routing should take place (inferred in HTML syntax)
+- routes: Array of routes or Application objects. _required_
```
Comment (TheMcMurder) [inline README.md:60]: I'm not in love with _required_ at the end but I wasn't sure if using JSDOC style syntax here was a good idea or not.

File: README.md [hunk 2]
```diff
@@ -5,7 +5,9 @@
 Layout engine for single-spa applications
 
 ## Project Goal
+
```
Comment (TheMcMurder) [inline README.md:8]: First draft of this I did in github's code editor so the prettier hook didn't run. This one I did locally so the prettier hook ran. 

The spacing and `'` -> `"` changes are from prettier

File: README.md [hunk 3]
```diff
@@ -33,16 +37,56 @@ const layoutEngine = constructLayoutEngine({routes, applications})
 start()
 ```
 
-
 ### Constructing Routes
+
 single-spa-layout supports two different sytanxes defining applications and routes at this time:
+
```
Comment (TheMcMurder) [inline README.md:43]: OPPORTUNITY TO USE `syntaxi` 🍷

File: README.md (hunk 1 above)
Comment (joeldenning) [reply README.md:60]: I'm renaming this to `children` in #21

File: README.md [hunk 4]
```diff
@@ -33,16 +37,56 @@ const layoutEngine = constructLayoutEngine({routes, applications})
...
+- base: base url where the router is active
+- containerEl: which cssSelector is where the routing should take place (inferred in HTML syntax)
+- routes: Array of routes or Application objects. _required_
+
+##### Route
+
+A routing point where all children will be conditionally rendered when the url matches. Supports the following options
+
+- path: string to evaluate against the URL. _required_
+- routes: Array of routes or Application objects. _required_
```
Comment (joeldenning) [inline README.md:67]: rename to `children`

File: README.md [hunk 5]
```diff
@@ -33,16 +37,56 @@ const layoutEngine = constructLayoutEngine({routes, applications})
 start()
 ```
 
-
 ### Constructing Routes
+
 single-spa-layout supports two different sytanxes defining applications and routes at this time:
```
Comment (joeldenning) [inline README.md:42]:
```suggestion
single-spa-layout supports two different syntaxes defining applications and routes at this time:
```

File: README.md [hunk 6]
```diff
@@ -33,16 +37,56 @@ const layoutEngine = constructLayoutEngine({routes, applications})
...
+
 - [JSX](#JSX) (compiles to JSON syntax)
```
Comment (joeldenning) [inline README.md:44]: I'm thinking we'll remove JSX in favor of HTML sent down in the HTML file. I think JSX will cause confusion in the following ways:

1. Can I call setState?
2. Is this react?
3. `className` vs `class`

Comment (TheMcMurder) [reply README.md:44]: Yeah I can do that in the next MR

**Pull request merged**